src/Symfony/Component/Form/Form.php

```diff
@@ -507,6 +508,10 @@ public function handleRequest($request = null)
      */
     public function submit($submittedData, $clearMissing = true)
     {
+        if ($submittedData instanceof Request) {
+            return $this->handleRequest($submittedData);
```

Comment:
ah there is a `return` in front of handleRequest. I didn't see that and this is what I thought is missing.